Sources/MapboxNavigation/NavigationViewController.swift

```diff
@@ -347,35 +343,35 @@ open class NavigationViewController: UIViewController, NavigationStatusPresenter
 
      The route may come directly from the completion handler of the [MapboxDirections](https://docs.mapbox.com/ios/api/directions/) framework’s `Directions.calculate(_:completionHandler:)` method, or it may be unarchived or created from a JSON object.
      
-     - parameter route: The route to navigate along.
+     - parameter routeResponse: `RouteResponse` object, containing selection of routes to follow.
      - parameter routeIndex: The index of the route within the original `RouteResponse` object.
      - parameter routeOptions: The route options used to get the route.
      - parameter navigationOptions: The navigation options to use for the navigation session.
      */
-    required public init(for route: Route, routeIndex: Int, routeOptions: RouteOptions, navigationOptions: NavigationOptions? = nil) {
+    required public init(for routeResponse: RouteResponse, routeIndex: Int, routeOptions: RouteOptions, navigationOptions: NavigationOptions? = nil) {
         if let options = navigationOptions {
             mapTileStore = options.tileStoreConfiguration.mapLocation
         }
         
-        super.init(nibName: nil, bundle: nil)
         
-        self.route = route
-        self.routeIndex = routeIndex
-        self.routeOptions = routeOptions
+        super.init(nibName: nil, bundle: nil)
+        self._routeResponse = routeResponse
+        self._routeIndex = routeIndex
+        self._routeOptions = routeOptions
         self.navigationOptions = navigationOptions
     }
     
     /**
      Initializes a `NavigationViewController` with the given route and navigation service.
      
-     - parameter route: The route to navigate along.
+     - parameter routeResponse: `RouteResponse` object, containing selection of routes to follow.
      - parameter routeIndex: The index of the route within the original `RouteResponse` object.
      - parameter routeOptions: the options object used to generate the route.
      - parameter navigationService: The navigation service that manages navigation along the route.
      */
-    convenience init(route: Route, routeIndex: Int, routeOptions: RouteOptions, navigationService service: NavigationService) {
+    convenience init(routeResponse: RouteResponse, routeIndex: Int, routeOptions: RouteOptions, navigationService service: NavigationService) {
```

Comment:
Now that NavigationService contains the full `indexedRouteResponse`, as well as the route options, do we need the application to pass in the route response, route index, and route options separately here, or is it sufficient to pass in just the navigation service?

Reply:
Do you mean a convenience init which would get these values from the custom navigation service provided? Sounds good to me.
But I think we still need to preserve this initializer. Otherwise we'll force users to explicitly instantiate the navigation service even if they are happy with the default one.